components/pages/explorer/Jbrowse/JbrowseWrapper.tsx

```diff
@@ -19,30 +19,204 @@
+import createArrangerFetcher from '@/components/utils/arrangerFetcher';
+import { Spinner } from '@/components/theme/icons';
+import ErrorNotification from '@/components/ErrorNotification';
+import {
+  JbrowseQueryNode,
+  JbrowseCompatibleFile,
+  JbrowseInput,
+  ScoreDownloadResult,
+  ScoreDownloadParams,
+} from './types';
+import { checkJbrowseCompatibility } from './utils';
 import JbrowseSelectedFilesTable from './JbrowseSelectedFilesTable';
 
+const { NEXT_PUBLIC_SCORE_API_URL } = getConfig();
+const arrangerFetcher = createArrangerFetcher({});
+
+// request data for jbrowse display and
+// score /download request to get signed URLs
+const jbrowseInputQuery = `
+query jbrowseInput($filters:JSON){
+  file {
+    hits (filters: $filters){
+      total 
+      edges {
+        node {
+          file_access
+          file_type
+          object_id
+          file {
+            name
+            size
+            index_file {
+              object_id
+              size
+            }
+          }
+        }
+      }
+    }
+  }
+}
+`;
+
+const baseScoreDownloadParams = {
+  external: 'true',
+  offset: '0',
+  'User-Agent': 'unknown',
+};
+
+const getScoreDownloadUrls = (type: 'file' | 'index', files: JbrowseCompatibleFile[]) =>
+  Promise.all(
+    files.map((file: JbrowseCompatibleFile) => {
+      const length = file[`${type}Size`].toString();
+      const object_id = file[`${type}Id`];
+
+      const scoreDownloadParams: ScoreDownloadParams = {
+        ...baseScoreDownloadParams,
+        length,
+        object_id,
+      };
+      const urlParams = new URLSearchParams(scoreDownloadParams).toString();
+
+      return fetch(
+        urlJoin(NEXT_PUBLIC_SCORE_API_URL, SCORE_API_DOWNLOAD_PATH, object_id, `?${urlParams}`),
+        {
+          headers: { accept: '*/*' },
+          method: 'GET',
+        },
+      ).then((response) => response.json());
+    }),
+  );
+
+const getUrlFromResult = (results: ScoreDownloadResult[], targetId: string) =>
+  find(results, { objectId: targetId })?.parts[0].url || '';
+
 const JbrowseWrapper = () => {
   const theme = useTheme();
+  const { selectedRows } = useTableContext({
+    callerName: 'Jbrowse Wrapper',
+  });
+  const [jbrowseEnabled, setJbrowseEnabled] = useState<boolean>(true);
+  const [jbrowseCompatibleFiles, setJbrowseCompatibleFiles] = useState<JbrowseCompatibleFile[]>([]);
+  const [jbrowseInput, setJbrowseInput] = useState<JbrowseInput[]>([]);
+  const [jbrowseLoading, setJbrowseLoading] = useState<boolean>(true);
+  const [jbrowseError, setJbrowseError] = useState<string>('');
+
+  const handleError = (error: Error) => {
+    setJbrowseEnabled(false);
+    setJbrowseError('Something went wrong.');
+    console.error(error);
+  };
+
+  useEffect(() => {
+    // check if any files in selectedRows are compatible with jbrowse.
+
+    setJbrowseLoading(true);
+
+    // fetch metadata from arranger for selected files
+    arrangerFetcher({
+      endpoint: 'graphql/TableDataQuery',
```

Comment:
That bit after the `graphql/` is not functional (i.e. can be customised) and I implemented it for declarative purposes, to differentiate which query is which.
```suggestion
      endpoint: 'graphql/JBrowseDataQuery',
```